hbase-shaded/hbase-shaded-testing-util/pom.xml

```diff
@@ -151,6 +151,7 @@
         <dependency>
           <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-common</artifactId>
+          <version>${hadoop.version}</version>
```

Comment:
We should add the version in dependencyManagement section in the parent pom?

Reply:
> We should add the version in dependencyManagement section in the parent pom?

I tried to add it to root pom but found some other confusion. 

We do have a hadoop-common in dependencyManagement: https://github.com/apache/hbase/blob/8e72cc6ee0415608d2d31e1c0c0175e8fc228d8d/pom.xml#L3918
But there seems to be a problem with this way of writing. Why do you need to specify both classifier and type? It seems that the correct jar cannot be located with these way. Not only hadoop-common, hadoop-hdfs also has this problem.

So if I'm correct, this seems to be a bug? Should we remove the classifier?

Reply:
<type>jar</type>

Reply:
```
<classifier>tests</classifier>
<type>jar</type>
```
and
```<type>test-jar</type>```
can both work, but
```
<classifier>tests</classifier>
<type>test-jar</type>
```
can't.

Reply:
Then let's do a cleanup in the dependencyManagement section?

Reply:
> Then let's do a cleanup in the dependencyManagement section?

I think so. If you don't mind, could we just merge this PR and https://github.com/apache/hbase/pull/5000 at first? I'd like to open another ticket for the new problem.

Reply:
No, we should not add version in the sub module's pom files, especially for hadoop, as we will depend on hadoop in multiple sub modules. Let's fix the hadoop-common version problem first, and then merge #5000, as the problem for #5000 only affects master branch.

Reply:
OK. Will push a new fix later.